test-utils/src/main/java/io/littlehorse/test/LHExtension.java

```diff
@@ -75,26 +75,34 @@ public void postProcessTestInstance(Object testInstance, ExtensionContext contex
         testContext.instrument(testInstance);
     }
 
+    @Override
+    public void preDestroyTestInstance(ExtensionContext context) {
+        ExtensionContext.Store store = getStore(context);
+        Object testInstance = context.getTestInstance().get();
+        TestContext testContext = store.get(LH_TEST_CONTEXT, TestContext.class);
+
+        for (String taskDef : testContext.discoverTaskDefNames(testInstance)) {
+            LHTaskWorker worker = (LHTaskWorker) store.get(taskDef);
+            worker.close();
+            store.remove(taskDef);
+        }
+    }
+
     private void maybeCreateTenantAndPrincipal(TestContext testContext) {
         if (testContext.getConfig().getTenantId() == null) {
             return;
         }
-        try {
-            Awaitility.await()
-                    .atMost(Duration.ofSeconds(25))
-                    .ignoreException(RuntimeException.class)
-                    .until(() -> {
-                        testContext
-                                .getLhClient()
-                                .putTenant(PutTenantRequest.newBuilder()
-                                        .setId(testContext.getConfig().getTenantId())
-                                        .build());
-                        return true;
-                    });
-        } catch (StatusRuntimeException ex) {
-            if (!ex.getStatus().getCode().equals(Status.Code.ALREADY_EXISTS)) {
-                throw ex;
-            }
-        }
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(25))
+                .ignoreExceptionsMatching(exn -> RuntimeException.class.isAssignableFrom(exn.getClass()))
+                .until(() -> {
+                    testContext
+                            .getLhClient()
+                            .withDeadlineAfter(2, TimeUnit.SECONDS)
```

Comment:
We have a weakness in our server where if we send a request before the server is fully rebalanced, it will hang until the default grpc deadline. @eduwercamacaro we should look at that this week. If I remove the `deadlineAfter` then the `StandaloneTestBootstrapper` wont start on my laptop (it works fine on the pipeline though because the smaller resources causes different sequencing).